src/libawkward/builder/StringBuilder.cpp

```diff
@@ -11,212 +13,181 @@
-    return out;
-  }
+const BuilderPtr StringBuilder::null() {
+  BuilderPtr out = OptionBuilder::fromvalids(options_, shared_from_this());
+  out.get()->null();
+  return out;
+}
 
-  const BuilderPtr
-  StringBuilder::boolean(bool x) {
-    BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
-    out.get()->boolean(x);
-    return out;
-  }
+const BuilderPtr StringBuilder::boolean(bool x) {
+  BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
+  out.get()->boolean(x);
+  return out;
+}
 
-  const BuilderPtr
-  StringBuilder::integer(int64_t x) {
-    BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
-    out.get()->integer(x);
-    return out;
-  }
+const BuilderPtr StringBuilder::integer(int64_t x) {
+  BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
+  out.get()->integer(x);
+  return out;
+}
 
-  const BuilderPtr
-  StringBuilder::real(double x) {
-    BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
-    out.get()->real(x);
-    return out;
-  }
+const BuilderPtr StringBuilder::real(double x) {
+  BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
+  out.get()->real(x);
+  return out;
+}
 
-  const BuilderPtr
-  StringBuilder::complex(std::complex<double> x) {
-    BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
-    out.get()->complex(x);
-    return out;
-  }
+const BuilderPtr StringBuilder::complex(std::complex<double> x) {
+  BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
+  out.get()->complex(x);
+  return out;
+}
 
-  const BuilderPtr
-  StringBuilder::datetime(int64_t x, const std::string& unit) {
-    BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
-    out.get()->datetime(x, unit);
-    return out;
-  }
+const BuilderPtr StringBuilder::datetime(int64_t x, const std::string &unit) {
+  BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
+  out.get()->datetime(x, unit);
+  return out;
+}
 
-  const BuilderPtr
-  StringBuilder::timedelta(int64_t x, const std::string& unit) {
-    BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
-    out.get()->timedelta(x, unit);
-    return out;
-  }
+const BuilderPtr StringBuilder::timedelta(int64_t x, const std::string &unit) {
+  BuilderPtr out = UnionBuilder::fromsingle(options_, shared_from_this());
+  out.get()->timedelta(x, unit);
+  return out;
+}
 
-  const BuilderPtr
-  StringBuilder::string(const char* x, int64_t length, const char* /* encoding */) {
-    if (length < 0) {
-      for (int64_t i = 0;  x[i] != 0;  i++) {
-        content_.append((uint8_t)x[i]);
-      }
+const BuilderPtr StringBuilder::string(const char *x, int64_t length,
+                                       const char * /* encoding */) {
+  if (length < 0) {
+    for (int64_t i = 0; x[i] != 0; i++) {
+      content_.append((uint8_t)x[i]);
     }
```

Comment:
Here's something that I don't expect it to preserve. For decades, I've been writing for loops like this:

```c++
for (int i = 0;  i < 10;  i++) {
```

rather than

```c++
for (int i = 0; i < 10; i++) {
```

The three parts of the for header are statements, separated by semicolons, and so I give them more room than comma-separated list data. But I think I'm the only person who does that—I never attempted to popularize the style.

So clang-format almost certainly doesn't have an option for it, and that's fine.